contents/performance/Performance.md

```diff
@@ -0,0 +1,104 @@
+
+If you are interested in learning more about Data Structures, you can learn more from the following resources:
+
+* [Geeksforgeeks.org](http://www.geeksforgeeks.org/data-structures/)
+* [Introduction to Algorithms (Book)](https://www.amazon.com/Introduction-Algorithms-3rd-MIT-Press/dp/0262033844)
+* [MIT online course](http://courses.csail.mit.edu/6.851/spring12/lectures/)
+* [topcoder.com](https://www.topcoder.com/community/data-science/data-science-tutorials/)
+
+There are more of such resources online. Just Google and you shall find.
+
+## Using Cache Friendly code
+
+Before we can talk about this, we need to know what the computer memory is. Computer memory has different components, [registers](https://en.wikipedia.org/wiki/Processor_register), [L1/L2/L3 cache](https://www.cs.umd.edu/class/fall2001/cmsc411/proj01/cache/cache.html), [RAM](https://en.wikipedia.org/wiki/Random-access_memory), and [disk](https://en.wikipedia.org/wiki/Hard_disk_drive) in order of their speed.
+
+> You can take a look at the extent of the difference of their access speeds [here](https://gist.github.com/jboner/2841832) <br>
+> You can also check out this [infographic](http://imgur.com/8LIwV4C)
+
+To understand this topic, you only need to know how the cache works. It is okay if you do not understand exactly how the other components, registers, RAM and disk work.
+
+The faster a cache is, the more expensive it is. This means that the L1 cache (a few kilobytes big) is smaller than the L2 cache (a few megabytes big) and so on, until the RAM (a few gigabytes big).
+
+Whenever data is requested, the computer will first look in the cache for the data. If it exists, this is known as a **cache hit**. If it does not exist, this is known as a **cache miss**. When a **cache miss** happens, a contiguous block of memory (not just the requested data) is retrieved and stored onto the cache. Because of this, we want remember these rules to make cache friendly codes:
+
+* Temporal Locality
+* Spatial Locality
+* Row/Column Major Order
+
+#### Temportal Locality
+
+This rule states that recently used memory will likely be used in the near future. This means that making the scope of your variables smaller helps with execution times, as it will likely result in less cache misses.
+
+#### Spatial Locality
+
+This rule states that memory stored near each other will likely be used in the near future. This means that using contiguous data structures, like arrays, helps improve the execution times. This is because the contiguous block of memory retrieved will likely contain the other memory that need to be used.
```

Comment:
This means that using contiguous data structures, ~~like~~ such as arrays, helps improve execution times.

You might want to make the last sentence clearer. "contain the other memory that need to be used" sounds quite weird